plugins/woocommerce-admin/client/inbox-panel/index.js

```diff
@@ -214,18 +261,32 @@ const InboxPanel = ( { showHeader = true } ) => {
 					}
 					return note;
 				} ),
-				isError: Boolean(
-					getNotesError( 'getNotes', [ INBOX_QUERY ] )
-				),
-				isResolvingNotes: isResolving( 'getNotes', [ INBOX_QUERY ] ),
+				isError: Boolean( getNotesError( 'getNotes', [ inboxQuery ] ) ),
 				isBatchUpdating: isNotesRequesting( 'batchUpdateNotes' ),
+				notesHaveResolved:
+					! isNotesRequesting( 'batchUpdateNotes' ) &&
+					hasFinishedResolution( 'getNotes', [ inboxQuery ] ),
 			};
-		}
+		} );
+
+	const allNotesHash = useMemo(
```

Comment:
Using simple hash computed from IDs to know when the notes results have changed.

Reply:
Having to use this feels like a bit of a code smell. Was there a specific reason you couldn't rely on the `notes` object reference being different?
I did notice that the `useSelect` above runs a `map` function inside of the `useSelect` which we want to avoid as it breaks memoization within the `useSelect` function. I do realize this was already the case prior, but might fix your issue with relying on the `notes` object and knowing when it changed or not.

Reply:
This was needed due to the memoization of the useSelect being broken by both the query as well as the `.map()` that you mention. This is fixed in https://github.com/woocommerce/woocommerce/pull/35003/commits/76a699601e389eacceb8f9d98b42e26d35683541.